mmtrack/models/backbones/mixformer_backbone.py

```diff
@@ -0,0 +1,699 @@
+                    **kwargs))
+        self.blocks = nn.ModuleList(blocks)
+
+    def init_weights(self):
+        if self.cls_token is not None:
+            trunc_normal_(self.cls_token, std=.02)
+
+        if self.init == 'xavier':
+            self.apply(self._init_weights_xavier)
+        else:
+            self.apply(self._init_weights_trunc_normal)
+
+    def _init_weights_trunc_normal(self, m):
+        if isinstance(m, nn.Linear):
+            logging.info('=> init weight of Linear from trunc norm')
+            trunc_normal_(m.weight, std=0.02)
+            if m.bias is not None:
+                logging.info('=> init bias of Linear to zeros')
+                nn.init.constant_(m.bias, 0)
+        elif isinstance(m, (nn.LayerNorm, nn.BatchNorm2d)):
+            nn.init.constant_(m.bias, 0)
+            nn.init.constant_(m.weight, 1.0)
+
+    def _init_weights_xavier(self, m):
+        if isinstance(m, nn.Linear):
+            logging.info('=> init weight of Linear from xavier uniform')
+            nn.init.xavier_uniform_(m.weight)
+            if m.bias is not None:
+                logging.info('=> init bias of Linear to zeros')
+                nn.init.constant_(m.bias, 0)
+        elif isinstance(m, (nn.LayerNorm, nn.BatchNorm2d)):
+            nn.init.constant_(m.bias, 0)
+            nn.init.constant_(m.weight, 1.0)
+
+    def forward(self, template, online_template, search):
+        """
+        :param template: (batch, c, 128, 128)
+        :param search: (batch, c, 320, 320)
+        :return:
+        """
+        # x = self.patch_embed(x)
+        # B, C, H, W = x.size()
+        template = self.patch_embed(template)
+        online_template = self.patch_embed(online_template)
+        t_B, t_C, t_H, t_W = template.size()
+        search = self.patch_embed(search)
+        s_B, s_C, s_H, s_W = search.size()
+
+        template = rearrange(template, 'b c h w -> b (h w) c').contiguous()
+        online_template = rearrange(online_template,
+                                    'b c h w -> b (h w) c').contiguous()
+        search = rearrange(search, 'b c h w -> b (h w) c').contiguous()
+        x = torch.cat([template, online_template, search], dim=1)
+
+        x = self.pos_drop(x)
+
+        for i, blk in enumerate(self.blocks):
+            x = blk(x, t_H, t_W, s_H, s_W)
+
+        # if self.cls_token is not None:
+        #     cls_tokens, x = torch.split(x, [1, H*W], 1)
+        template, online_template, search = torch.split(
+            x, [t_H * t_W, t_H * t_W, s_H * s_W], dim=1)
+        template = rearrange(
+            template, 'b (h w) c -> b c h w', h=t_H, w=t_W).contiguous()
+        online_template = rearrange(
+            online_template, 'b (h w) c -> b c h w', h=t_H,
+            w=t_W).contiguous()
+        search = rearrange(
+            search, 'b (h w) c -> b c h w', h=s_H, w=s_W).contiguous()
+
+        return template, online_template, search
+
+    def forward_test(self, search):
```

Comment:
Suggest add docstring for these funct so that user can konw the shape of inputs and outputs easily.